### PR TITLE
fix: core fixes

### PR DESCRIPTION
**Pull request by sattvikc**
## Summary of change

- Returning `tenantId` in some of the APIs for session
- Making argon and bcrypt properties config yaml only configs
- Making `ip_allow_regex` and `ip_deny_regex` protected props for SaaS
- Adding `hello` to disallowed tenant ids
- Creation of app has all recipes enabled by default while creating tenant does not

## Related issues

- Link to issue1 here
- Link to issue1 here

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your
changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here
highlighting the necessary changes)

## Checklist for important updates

- [ ] Changelog has been updated
    - [ ] If there are any db schema changes, mention those changes clearly
- [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
- [ ] `pluginInterfaceSupported.json` file has been updated (if needed)
- [ ] Changes to the version if needed
    - In `build.gradle`
- [ ] If added a new paid feature, edit the `getPaidFeatureStats` function in FeatureFlag.java file
- [ ] Had installed and ran the pre-commit hook
- [ ] If there are new dependencies that have been added in `build.gradle`, please make sure to add them
  in `implementationDependencies.json`.
- [ ] Update function `getValidFields` in `io/supertokens/config/CoreConfig.java` if new aliases were added for any core config (similar to the `access_token_signing_key_update_interval` config alias).
- [ ] Issue this PR against the latest non released version branch.
    - To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the
      latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    - If no such branch exists, then create one from the latest released branch.

## Remaining TODOs for this PR

- [ ] Item1
- [ ] Item2

### CODE REVIEW

File: src/main/java/io/supertokens/webserver/api/session/SessionAPI.java [hunk 1]
```diff
@@ -123,6 +123,8 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
 
             JsonObject result = sessionInfo.toJsonObject();
 
+            result.get("session").getAsJsonObject().remove("tenantId");
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/webserver/api/session/SessionAPI.java:126]: keep it

File: src/main/java/io/supertokens/webserver/api/multitenancy/ListTenantsAPI.java [hunk 2]
```diff
@@ -59,8 +60,18 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IO
             TenantConfig[] tenantConfigs = Multitenancy.getAllTenantsForApp(tenantIdentifierWithStorage.toAppIdentifier(), main);
             JsonArray tenantsArray = new JsonArray();
 
+            boolean shouldProtect = shouldProtectProtectedConfig(req);
             for (TenantConfig tenantConfig : tenantConfigs) {
-                tenantsArray.add(tenantConfig.toJson(shouldProtectDbConfig(req), tenantIdentifierWithStorage.getStorage()));
+                JsonObject tenantConfigJson = tenantConfig.toJson(shouldProtect,
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/webserver/api/multitenancy/ListTenantsAPI.java:65]: do this in the plugin interface

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: src/main/java/io/supertokens/webserver/api/session/SessionAPI.java (hunk 1 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/webserver/api/session/SessionAPI.java:126]: Done

File: src/main/java/io/supertokens/webserver/api/multitenancy/ListTenantsAPI.java (hunk 2 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/webserver/api/multitenancy/ListTenantsAPI.java:65]: Done

**Pull request merged**